src/http-api/errors.ts

```diff
@@ -49,12 +52,14 @@ export class MatrixError extends HTTPError {
      * information specific to the standard Matrix error response.
      * @param errorJson - The Matrix error JSON returned from the homeserver.
      * @param httpStatus - The numeric HTTP status code given
+     * @param httpHeaders - The HTTP response headers given
      */
     public constructor(
         errorJson: IErrorJson = {},
         public readonly httpStatus?: number,
         public url?: string,
         public event?: MatrixEvent,
+        public readonly httpHeaders?: Headers,
```

Comment:
Ah true, this shadows `HttpError`'s member, which was not my intention.

`httpStatus` probably shouldn't be shadowing either, but changing it could break existing behaviour, so I'm inclined for it to be tackled it in another PR.  (I'm also not a fan of how `errcode` is redundant & separate from `data.errcode`.)

Reply:
> httpStatus probably shouldn't be shadowing either, but changing it could break existing behaviour

I don't think it can?  I don't think it *actually* shadows `HttpError`'s member -- it just *looks* like it does.

Happy for you to punt that change to another PR though, in any case.

Reply:
Yes, you're right, there's no shadowing here after all -- a parameter property named the same as a base class property ends up referring to the same property, not a distinct one.  I was stuck in a C++ mindset where base class members are distinct from subclass members.

Reply:
6a55da0a2